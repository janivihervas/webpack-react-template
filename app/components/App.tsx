import React, {PureComponent} from "react"
import "./App.scss"

export interface Props {
  message: string
}

export default class App extends PureComponent<Props> {
  render() {
    return (
      <div className="app">
        <h1>{this.props.message}</h1>
        <p>lololo</p>
      </div>
    )
  }
}
