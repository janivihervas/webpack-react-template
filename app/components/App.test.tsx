import React from "react"
import TestUtils from "react-addons-test-utils"
import {describe, it} from "mocha"
import {assert} from "chai"

import App from "./App.jsx"
import {assertReactElements} from "../helpers"

describe("App", () => {
  const renderer = TestUtils.createRenderer()

  it("should render correctly", () => {
    const fn = () => {}

    renderer.render(<App mounted={fn} />)
    const el = renderer.getRenderOutput()

    assertReactElements(
      el,
      <div>
        <h1 />
      </div>,
    )
  })

  it("should call this.props.mounted() on componentDidMount", () => {
    let called = false
    const fn = () => {
      called = true
    }

    renderer.render(<App mounted={fn} />)

    const instance = renderer._instance._instance

    assert(!called)
    instance.componentDidMount()
    assert(called)
  })
})
