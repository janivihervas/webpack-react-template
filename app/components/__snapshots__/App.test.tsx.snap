// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render 1`] = `
<div
  className="app"
>
  <h1>
    Hello World!
  </h1>
</div>
`;
